src/InputGroup/InputGroup.test.js

```diff
@@ -215,100 +139,54 @@ describe('<InputGroup />', () => {
+    };
 
-            expect(
-                element.getDOMNode().attributes['data-sample'].value
-            ).toBe('Sample');
+    describe('Default Rendering', () => {
+        let element = setup({
+            children: 'foo' //prop children is required
         });
 
-        test('should allow props to be spread to the InputGroup component for addonPos before\'s (type text) input element', () => {
-            const element = mount(<InputGroup addonPos='before' inputProps={{ 'data-sample': 'Sample' }} />);
-
+        it('should render', () => {
             expect(
-                element.find('input').getDOMNode().attributes['data-sample'].value
-            ).toBe('Sample');
+                element
+            ).toBeDefined();
         });
 
-        test('should allow props to be spread to the InputGroup component for type number', () => {
-            const element = mount(<InputGroup data-sample='Sample' inputType='number' />);
-
+        it('should render a div tag by default', () => {
             expect(
-                element.getDOMNode().attributes['data-sample'].value
-            ).toBe('Sample');
+                element.getDOMNode().tagName
+            ).toBe('DIV');
         });
 
-        test('should allow props to be spread to the InputGroup component for type number\'s input element', () => {
-            const element = mount(<InputGroup inputProps={{ 'data-sample': 'Sample' }} inputType='number' />);
-
+        it('should have a default class of "fd-input-group"', () => {
             expect(
-                element.find('input').getDOMNode().attributes['data-sample'].value
-            ).toBe('Sample');
+                element.getDOMNode().className
+            ).toContain('fd-input-group');
         });
+    });
 
-        test('should allow props to be spread to the InputGroup component for type number\'s up button element', () => {
-            const element = mount(<InputGroup inputType='number' numberUpButtonProps={{ 'data-sample': 'Sample' }} />);
+    describe('Rendering with Props', () => {
+        it('should not throw error if child.type is undefined', () => {
+            let element = setup({
+                children: <div>This is a test</div>
+            });
 
-            expect(
-                element.find('button.sap-icon--slim-arrow-up').getDOMNode().attributes['data-sample'].value
-            ).toBe('Sample');
+            expect(() => element).not.toThrow();
         });
 
-        test('should allow props to be spread to the InputGroup component for type number\'s down button element', () => {
-            const element = mount(<InputGroup inputType='number' numberDownButtonProps={{ 'data-sample': 'Sample' }} />);
+        it('should allow data attribute to be passed to span element', () => {
+            const element = setup({
+                children: [<InputGroup.Input />, <InputGroup.Addon>@</InputGroup.Addon>],
+                'data-sample': 'Sample'
```

Comment:
The way this is setup, shouldn't this `data-sample` be being passed to the input group, not the addon?

Reply:
ah ok - the more I look at this - the test is right - just the description should be updated.